tools/generator/cmd/track2/common/fileProcessor.go

```diff
@@ -0,0 +1,222 @@
+			err = os.Remove(filepath.Join(path, file.Name()))
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// replace all commitid in autorest.md files
+func ReplaceCommitid(path string, commitid string) error {
+	log.Printf("Replacing commitid from autorest.md ...")
+	b, err := ioutil.ReadFile(path)
+	if err != nil {
+		return err
+	}
+
+	lines := strings.Split(string(b), "\n")
+	for i, line := range lines {
+		if strings.HasPrefix(line, autorest_md_swagger_url_prefix) {
+			lines[i] = lines[i][:len(autorest_md_swagger_url_prefix)] + commitid + lines[i][len(autorest_md_swagger_url_prefix)+len(commitid):]
+		}
+	}
+
+	err = ioutil.WriteFile(path, []byte(strings.Join(lines, "\n")), 0644)
+	return err
+}
+
+func GetLatestVersion(packageRootPath string) (string, error) {
+	b, err := ioutil.ReadFile(filepath.Join(packageRootPath, "autorest.md"))
+	if err != nil {
+		return "", err
+	}
+
+	lines := strings.Split(string(b), "\n")
+	for _, line := range lines {
+		if strings.HasPrefix(line, autorest_md_module_version_prefix) {
+			return strings.TrimSuffix(strings.TrimSuffix(line[len(autorest_md_module_version_prefix):], "\n"), "\r"), nil
+		}
+	}
+
+	return "", fmt.Errorf("cannot parse version from autorest.md")
+}
+
+func ReplaceVersion(packageRootPath string, newVersion string) error {
+	path := filepath.Join(packageRootPath, "autorest.md")
+	b, err := ioutil.ReadFile(path)
+	if err != nil {
+		return err
+	}
+
+	lines := strings.Split(string(b), "\n")
+	for i, line := range lines {
+		if strings.HasPrefix(line, autorest_md_module_version_prefix) {
+			lines[i] = lines[i][:len(autorest_md_module_version_prefix)] + newVersion + "\n"
+			break
+		}
+	}
+
+	err = ioutil.WriteFile(path, []byte(strings.Join(lines, "\n")), 0644)
+	return err
+}
+
+func CalculateNewVersion(changelog *model.Changelog, packageRootPath string) (string, error) {
+	version, err := GetLatestVersion(packageRootPath)
+	if err != nil {
+		return "", err
+	}
+
+	versions := strings.Split(version, ".")
+	major, err := strconv.Atoi(versions[0])
+	if err != nil {
+		return "", fmt.Errorf("wrong latest version")
+	}
+	minor, err := strconv.Atoi(versions[1])
+	if err != nil {
+		return "", fmt.Errorf("wrong latest version")
+	}
+	patch, err := strconv.Atoi(versions[2])
+	if err != nil {
+		return "", fmt.Errorf("wrong latest version")
+	}
+	log.Printf("Lastest version is: %d.%d.%d", major, minor, patch)
+
+	if major == 0 {
+		// preview version calculation
+		if changelog.HasBreakingChanges() {
+			minor += 1
+			patch = 0
+		} else {
+			patch += 1
+		}
+	} else {
+		// release version calculation
+		if changelog.HasBreakingChanges() {
+			major += 1
+			minor = 0
+			patch = 0
+		} else if changelog.Modified.HasAdditiveChanges() {
+			minor += 1
+			patch = 0
+		} else {
+			patch += 1
+		}
+	}
+
+	log.Printf("New version is: %d.%d.%d", major, minor, patch)
+	return fmt.Sprintf("%d.%d.%d", major, minor, patch), nil
```

Comment:
We actually have a dependency `github.com/MasterMind/semver` to deal semantic versions, we can leverage on that